diff Naur a/conf/layer.conf b/conf/layer.conf
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -10,14 +10,14 @@ BBFILES += " \
 "
 
 BBFILE_COLLECTIONS += "gnk"
-BBFILE_PATTERN_userboard = "^${LAYERDIR}/"
-BBFILE_PRIORITY_userboard = "20"
+BBFILE_PATTERN_gnk = "^${LAYERDIR}/"
+BBFILE_PRIORITY_gnk = "20"
 
 # The .bbappend and .bb files are included if the respective layer
 # collection is available.
-BBFILES += "${@' '.join('${LAYERDIR}/dynamic-layers/%s/*/*.bbappend' % layer \
-               for layer in BBFILE_COLLECTIONS.split())}"
-BBFILES += "${@' '.join('${LAYERDIR}/dynamic-layers/%s/*/*.bb' % layer \
-               for layer in BBFILE_COLLECTIONS.split())}"
+#BBFILES += "${@' '.join('${LAYERDIR}/dynamic-layers/%s/*/*.bbappend' % layer \
+#               for layer in BBFILE_COLLECTIONS.split())}"
+#BBFILES += "${@' '.join('${LAYERDIR}/dynamic-layers/%s/*/*.bb' % layer \
+#               for layer in BBFILE_COLLECTIONS.split())}"
 
 LAYERSERIES_COMPAT_gnk = "dunfell"
diff Naur a/recipes-bsp/flash-writer/flash-writer.bbappend b/recipes-bsp/flash-writer/flash-writer.bbappend
--- a/recipes-bsp/flash-writer/flash-writer.bbappend
+++ b/recipes-bsp/flash-writer/flash-writer.bbappend
@@ -1,6 +1,12 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 
-SRC_URI_append = " \
+SRC_URI_append_gnk-rzg2l = " \
+	file://makefile \
+	file://0001-GREENPAK_RZG2L-add-DDR4-1G-IS43QR16512A.patch \
+	file://0001-V2L-DDR-add-DDR4-1G-IS43QR16512A.patch \
+"
+
+SRC_URI_append_gnk-rzv2l = " \
 	file://makefile \
 	file://0001-GREENPAK_RZG2L-add-DDR4-1G-IS43QR16512A.patch \
 	file://0001-V2L-DDR-add-DDR4-1G-IS43QR16512A.patch \
diff Naur a/recipes-bsp/trusted-firmware-a/trusted-firmware-a.bbappend b/recipes-bsp/trusted-firmware-a/trusted-firmware-a.bbappend
--- a/recipes-bsp/trusted-firmware-a/trusted-firmware-a.bbappend
+++ b/recipes-bsp/trusted-firmware-a/trusted-firmware-a.bbappend
@@ -2,7 +2,14 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
 
 #COMPATIBLE_MACHINE_rzv2l_rzg2l = "(gnk-rzg2l|gnk-rzv2l)"
 
-SRC_URI_append = " \
+SRC_URI_append_gnk-rzg2l = " \
+	file://0001-add-DDR4-1GB-IS43QR16512A.patch \
+	file://0002-add-SPI-Flash-IS25WP512M.patch \
+	file://0003-enable-SPI-Flash-IS25WP512M.patch \
+	file://0004-GNK_V2L-add-DDR4-1GB-IS43QR16512A.patch \
+"
+
+SRC_URI_append_gnk-rzv2l = " \
 	file://0001-add-DDR4-1GB-IS43QR16512A.patch \
 	file://0002-add-SPI-Flash-IS25WP512M.patch \
 	file://0003-enable-SPI-Flash-IS25WP512M.patch \
diff Naur a/recipes-bsp/u-boot/u-boot_2021.10.bbappend b/recipes-bsp/u-boot/u-boot_2021.10.bbappend
--- a/recipes-bsp/u-boot/u-boot_2021.10.bbappend
+++ b/recipes-bsp/u-boot/u-boot_2021.10.bbappend
@@ -1,6 +1,14 @@
 FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
-SRC_URI += "file://0001-support-gnk-board-dts.patch \
+
+SRC_URI_append_gnk-rzg2l = " \
+	file://0001-support-gnk-board-dts.patch \
+	file://0002-support-gnk-board-config.patch \
+	file://0001-gnk-g2l-support-smarc-rzg2l-gnk-board.patch \
+"
+
+SRC_URI_append_gnk-rzv2l = " \
+	file://0001-support-gnk-board-dts.patch \
 	file://0002-support-gnk-board-config.patch \
 	file://0001-gnk-g2l-support-smarc-rzg2l-gnk-board.patch \
-	"
+"
 
diff Naur a/recipes-rzg2l/trusted-firmware-a/trusted-firmware-a.bbappend b/recipes-rzg2l/trusted-firmware-a/trusted-firmware-a.bbappend
--- a/recipes-rzg2l/trusted-firmware-a/trusted-firmware-a.bbappend
+++ b/recipes-rzg2l/trusted-firmware-a/trusted-firmware-a.bbappend
@@ -10,9 +10,10 @@ PMIC_EXTRA_FLAGS_gnk-rzg2l = "BOARD=gnk-rzg2l"
 FLASH_ADDRESS_BL2_BP_gnk-rzg2l = "00000"
 FLASH_ADDRESS_FIP_gnk-rzg2l = "1D200"
 
-SRC_URI_append = " \
+SRC_URI_append_gnk-rzg2l = " \
 	file://rz_board.mk \
 "
+
 do_addboards () {
 	mkdir -p ${S}/plat/renesas/rz/board/gnk-rzg2l
 	cp -fv ${WORKDIR}/rz_board.mk ${S}/plat/renesas/rz/board/gnk-rzg2l
diff Naur a/recipes-rzv2l/trusted-firmware-a/trusted-firmware-a.bbappend b/recipes-rzv2l/trusted-firmware-a/trusted-firmware-a.bbappend
--- a/recipes-rzv2l/trusted-firmware-a/trusted-firmware-a.bbappend
+++ b/recipes-rzv2l/trusted-firmware-a/trusted-firmware-a.bbappend
@@ -10,7 +10,7 @@ PMIC_EXTRA_FLAGS_gnk-rzv2l = "BOARD=gnk-rzv2l"
 FLASH_ADDRESS_BL2_BP_gnk-rzv2l = "00000"
 FLASH_ADDRESS_FIP_gnk-rzv2l = "1D200"
 
-SRC_URI_append = " \
+SRC_URI_append_gnk-rzv2l = " \
 	file://rz_board.mk \
 "
 do_addboards () {
